### ISSUE TITLE
XSS - Javascript enabled

### ISSUE THREAD

**Issue by Neosama** [wereturtle/ghostwriter#594]
**Platform:**
 - OS and version [e.g. Ubuntu 18.04, Windows 10, MacOS Sierra]: *Debian 10*
 - If Linux, Desktop Environment [e.g., Gnome, Plasma, Cinnamon, etc.]: *Gnome*

**ghostwriter installation (please complete the following information):**
 - ghostwriter version [see Help->About in menu bar]: *v1.7.4*

**Describe the bug**
The text field interprets the html code and javascript is enabled, this can be exploited by an attacker.

**To Reproduce**
1. Write for exemple : `<script>alert(1)</script>`
2. Enable preview html

**Expected behavior**
Disabling/filtering javascript seems safer, the js code would not be executed.

**Screenshots**
![2021-05-06 11-40-52](https://user-images.githubusercontent.com/27860757/117277233-f5915280-ae5f-11eb-9baa-c764eb944cab.png)

**Comment by matthiasbeyer** [comment]
Do I understand correctly that this is _only_ exploitable if preview mode is on? If I do not enable it, JS is not executed?

**Comment by Neosama** [comment]
Yes, if you edit in text mode it does not execute.

**Comment by abenson** [comment]
Can you test this on a more recent version of ghostwriter? 1.7.4 is quite old and I can't get it to behave the same on current versions.

**Comment by Neosama** [comment]
With version v1.8.1, <script> no longer works, but other tags do.
For exemple : `<a href="javascript:alert(123)" onmouseover="alert(123)">TEST</a>` or `<iframe onload="alert(1)">`
![Capture d’écran du 2021-05-06 16-46-56](https://user-images.githubusercontent.com/27860757/117319236-66019900-ae8b-11eb-8c35-9516c9f9c141.png)

**Comment by abenson** [comment]
This one seems to be fixed in the latest master, as well.

**Comment by Neosama** [comment]
I understand, many markdown editors are in the same situation.

However the execution of JS code without user intervention is now blocked. (version 2.0.0-rc5) (from what I could test, I may have missed something) I don't think code execution via a clickable link is critical. An alert warning the user that the link contains code is possible, with a detection of some strings like `javascript:`. But is it really necessary? This same link can lead to a malicious site that can execute all the JavaScript code it wants.

In summary, no big security problem (the Chrome application on Android has the same behavior). Possibly an alert before the link is opened with its display to let the user determine if the link is safe or not.

**Thanks for all the work done, I use ghostwriter a lot, I find it excellent.**
